lib/logger/lib/logger.ex

```diff
@@ -479,18 +479,19 @@ defmodule Logger do
   Use this function only when there is a need to
   explicitly avoid embedding metadata.
   """
-  @spec bare_log(level, message | (() -> message), Keyword.t) ::
+  @spec bare_log(level, message | (() -> message) | (() -> {message, Keyword.t}), Keyword.t) ::
```

Comment:
I think we can simplify this spec with: `(() -> message | {message, Keyword.t})`.